### DISCUSSION TITLE
Cron jobs getting 308 status code

### DISCUSSION THREAD

**Discussion by inavac182** [vercel/community#4448] [Help] [answered]
### Summary

Hey Vercel Team,

I started a project and added my custom domain, when I'm trying to set up the cron jobs they seem to be running against the deployment host rather than the custom domain and seems like this is causing them to get a 308 status code back? Is there anything that I could change in my set up to fix this?

Please let me know if you need any more info.
Thanks!

### Steps to Reproduce

Project domains:
<img width="532" alt="Screenshot 2023-10-11 at 2 22 54 PM" src="https://github.com/vercel/community/assets/16787893/05f48266-f11f-4f46-8a14-ad98d5f572a3">

Logs for the cron job running against the deployment host and getting 308 status:
<img width="886" alt="Screenshot 2023-10-11 at 2 23 23 PM" src="https://github.com/vercel/community/assets/16787893/8e9725e5-92c4-49d9-9e26-673b4b027ec7">

Logs if I access directly the path `/api/testcron` in from both domains, custom and deployment:
<img width="773" alt="Screenshot 2023-10-11 at 2 24 11 PM" src="https://github.com/vercel/community/assets/16787893/061134cc-6591-42bb-8400-438d0e1ee5ea">
<img width="728" alt="Screenshot 2023-10-11 at 7 55 26 PM" src="https://github.com/vercel/community/assets/16787893/166d9912-9b06-4f91-85e1-fb3989029cd8">

In my case these are the steps I followed to get to this issue:
1. Create a project
2. Set up a custom domain
3. Add a cron job
4. The executions get 308 status code

**Comment by z4nr34l** [comment]
308 is temporary redirection - please check your middleware and next.config if you not make any redirect that can affect those endpoints (most often it's caused by authorization).

The problem is not existing in clean Next.js repository, if above solution didn't worked please provide us with [minimal reproducible example](https://vercel.com/guides/creating-a-minimal-reproducible-example)

**Reply by inavac182** [reply]
Hey @z4nr34l, I was not able to find a root cause within the middleware, I'll proceed to create a reproducible example, a question I got about this is if is expected that cron jobs run against Preview environment rather than Production?

<img width="1728" alt="Screenshot 2023-10-13 at 5 57 43 PM" src="https://github.com/vercel/community/assets/16787893/1e47ce5d-1f3c-47ff-a27d-bb8302ed1fbc">

In the logs I don't see any execution over Production environment, all requests seem to be in Preview.

**Reply by inavac182** [reply]
I just created a Next JS project with a cron job and a custom domain to test this.

Domain: https://devxican.com/
Repo: https://github.com/inavac182/cron-issue-next-vercel 
Cron endpoint: https://devxican.com/api/testcron/

**Reply by inavac182** [reply]
AH, I just found the issue, the trailing `/` in the cron path

**Reply by inavac182** [reply]
I still have the question if the cron jobs should run against the Preview environment, but after removing the trailing `/` now jobs are working as expected, which was the main issue I was facing.

**Comment by inavac182** [comment, marked as answer]
For anybody that faces this issue, the solution is pretty simple, in the `vercel.json` make sure the paths doesn't include a trailing `/` as when adding this there is a redirect happening and as cron jobs doesn't follow redirects then the status of the cron execution is 308.

<img width="770" alt="Screenshot 2023-10-13 at 6 30 08 PM" src="https://github.com/vercel/community/assets/16787893/de4291bf-b3b7-4c8c-b981-573a5b9494bf">

Removing this trailing `/` fixes the issue for the 308 status.
